main.R

```diff
@@ -702,5 +712,5 @@ if (pacta.data.validation::validate_intermediate_scenario_output(geco_2022)) {
     fs::path(scenario_preparation_outputs_path, "geco_2022.csv")
   )
 } else {
-  rlang::abort("GECO 2022 data is not valid.")
+  logger::log_error("GECO 2022 data is not valid.")
```

Comment:
```suggestion
  logger::log_error("GECO 2022 data is not valid.")
  stop("GECO 2022 data is not valid.")
```

Reply:
Hmm this seems pretty verbose. Is there no way to do both at the same time?

Reply:
if you invoke `log_errors()`, it adds a `trace()` to `stop()`.
`log_warnings()` is similar.

But as a pattern, I like to have the "useful" messages in `log_warn`/`log_error`, since having access to `glue` make constructing those easier, and then the `stop()` or `warning()` is there in case the user has set the logging threshold past WARN (and to allow for normal control flow stuff, like `tryCatch`), but with a simple static string.


https://daroczig.github.io/logger/reference/log_errors.html

Reply:
In this case, I think we need to change the condition in the `if` to be `inherits("logical")`, since `pacta.data.validation::validate_intermediate_scenario_output` outputs an object of class [`AssertCollection`](https://mllg.github.io/checkmate/reference/AssertCollection.html). and then we  could emit those assertion failures in the `log_error`, and keep the same `stop`.

Reply:
NIT: In general, I prefer using `rlang::warn` and `rlang::abort` over their base R equivalents, because they also have access to `glue`

(This is the paradigm I will implement for error handling in `pacta.scenario.data.preparation`)

Reply:
To clarify: I don't have any problems with using `rlang::abort()`/`warn()` here. The if condition might be a problem, since I don't know what happens whe you have an `AssertCollection` as a condition.